homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,236 @@
+        self._state_led = data["state_led"]
+        self._follow_move = data["follow_move"]
+        self._alarm_notify = data["alarm_notify"]
+        self._alarm_sound_mod = data["alarm_sound_mod"]
+        self._encrypted = data["encrypted"]
+        self._local_ip = data["local_ip"]
+        self._detection_sensibility = data["detection_sensibility"]
+        self._device_sub_category = data["device_sub_category"]
+        self._local_rtsp_port = data["local_rtsp_port"]
+
+    async def async_added_to_hass(self):
+        """Subscribe to ffmpeg and add camera to list."""
+        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+        entities = self.hass.data.setdefault(EZVIZ_DATA, {}).setdefault(ENTITIES, [])
+        entities.append(self)
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return the Ezviz-specific camera state attributes."""
+        return {
+            "serial": self._serial,
+            "name": self._name,
+            "status": self._status,
+            "device_sub_category": self._device_sub_category,
+            "privacy": self._privacy,
+            "audio": self._audio,
+            "ir_led": self._ir_led,
+            "state_led": self._state_led,
+            "follow_move": self._follow_move,
+            "alarm_notify": self._alarm_notify,
+            "alarm_sound_mod": self._alarm_sound_mod,
+            "encrypted": self._encrypted,
+            "local_ip": self._local_ip,
+            "detection_sensibility": self._detection_sensibility,
+        }
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._status
+
+    @property
+    def brand(self):
+        """Return the camera brand."""
+        return "Ezviz"
+
+    @property
+    def supported_features(self):
+        """Return supported features."""
+        if self._rtsp_stream:
+            return SUPPORT_STREAM
+        return 0
+
+    @property
+    def model(self):
+        """Return the camera model."""
+        return self._device_sub_category
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._status
+
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
+
+    async def async_camera_image(self):
+        """Return a frame from the camera stream."""
+        ffmpeg = ImageFrame(self._ffmpeg.binary, loop=self.hass.loop)
+
+        image = await asyncio.shield(
+            ffmpeg.get_image(self._rtsp_stream, output_format=IMAGE_JPEG,)
+        )
+        return image
+
+    async def stream_source(self):
+        """Return the stream source."""
+        if self._local_rtsp_port:
+            rtsp_stream_source = "rtsp://{}:{}@{}:{}".format(
```

Comment:
Why do we need to create a new stream source?

Reply:
I thought this was needed to allow "stream:" in configurations ?

Reply:
Unless some part of the address changes during entity lifetime, we can still use `self._rtsp_stream` that we built before creatingtthe entity, right?

Reply:
the local IP is indeed a dynamic attribute. After the DHCP Lease, the camera itself can be given another IP.
tl;dr: I wish I could have had this attribute on my other HA-compatible cameras, as I had to dedicate static IPs to them with a view to getting a proper local rtsp stream

Reply:
Ok.